test/integration/smoke/test_direct_download.py

```diff
@@ -269,6 +277,21 @@ def createServiceOffering(self, name, type, tags):
             tags=tags
         )
 
+    def deployVM(self, offering) :
+        if self.zone.networktype == 'Basic' :
+            vm = VirtualMachine.create(
+                self.apiclient,
+                self.services["virtual_machine"],
+                serviceofferingid=offering.id
+            )
```

Comment:
this should add
```suggestion
            vm = VirtualMachine.create(
                self.apiclient,
                self.services["virtual_machine"],
                serviceofferingid=offering.id
            )
            self.cleanup.append(vm)
```
and not outside the method.
Same for createServiceOffering